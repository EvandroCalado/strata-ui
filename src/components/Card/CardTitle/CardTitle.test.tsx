import { render, screen } from '@testing-library/react';
import { CardTitle } from '.';

describe('<CardTitle />', () => {
  it('should render a CardTitle without icon', () => {
    render(<CardTitle>valid name</CardTitle>);

    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument();
    expect(screen.getByText(/valid name/i)).toBeInTheDocument();
    expect(screen.queryByTestId('icon')).not.toBeInTheDocument();
  });

  it('should render a snapshot', () => {
    const { container } = render(<CardTitle icon>valid name</CardTitle>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
