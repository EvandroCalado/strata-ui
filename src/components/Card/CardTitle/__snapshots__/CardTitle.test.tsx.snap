// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<CardTitle /> > should render a snapshot 1`] = `
<h1
  class="my-4 flex items-center justify-between text-xl font-semibold"
>
  valid name
  <svg
    class="lucide lucide-arrow-up-right"
    data-testid="icon"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 7h10v10"
    />
    <path
      d="M7 17 17 7"
    />
  </svg>
</h1>
`;
